app/moves/middleware.js

```diff
@@ -116,24 +115,46 @@ const movesMiddleware = {
           return moveService
             .getMovesCount({ dateRange, status: moveType.filter, locationId })
             .then(count => {
-              const { label } = moveType
               return {
-                label,
+                ...moveType,
                 value: count,
                 active: moveType.filter === req.params.status,
                 href: `${req.baseUrl}/${period}/${date}${
                   locationId ? '/' + locationId : ''
                 }/${moveType.filter}`,
               }
             })
-            .then(presenters.moveTypesToFilterComponent)
         })
       )
       next()
     } catch (error) {
       next(error)
     }
   },
+  setTotalMoves: (req, res, next) => {
+    const totalMoves = {
+      label: 'moves::dashboard.filter.total',
+      filter: 'total',
+      href: find(res.locals.moveTypeNavigation, { filter: 'proposed' }).href,
+    }
+    totalMoves.value = res.locals.moveTypeNavigation.reduce(
+      (accumulator, moveType) => {
+        return (accumulator += moveType.value)
+      },
+      0
+    )
+    res.locals.moveTypeNavigation = reject(res.locals.moveTypeNavigation, {
```

Comment:
I think it might be cleaner to use a separate variable name here rather than override `moveTypeNavigation`.

It just feels less dangerous that overriding the value of something further down the middleware chain. It also feels like for this instance its not really a navigation element anymore anyway. It's more of a summary of proposed moves when being used on the dashboard.